### PR TITLE
overlord,timings,daemon: separate timings from overlord/state

### PR DESCRIPTION
**Pull request by pedronis**
There are places now that import timings that shouldn't indirectly get overlord/state as well.

timings strictly doesn't need state in an essential way, use an interface to mediate between the two instead.

The cost of this is having to move the timings helpers specific to Change/Task to overlord/state itself.

LinkChange was defined but unused, start using the moved/renamed TagTimingsWithChange.

### CODE REVIEW

File: timings/state.go [hunk 1]
```diff
@@ -116,13 +115,21 @@ func flattenRecursive(data *rootTimingsJSON, timings []*Span, nestLevel int, max
 	}
 }
 
-// Save appends Timings data to the "timings" list in the state and purges old timings, ensuring
-// that up to MaxTimings are kept. Timings are only stored in state if their duration is greater
-// than or equal to DurationThreshold.
-// It's responsibility of the caller to lock the state before calling this function.
-func (t *Timings) Save(st *state.State) {
+// A GetSaver helps storing Timings (ignoring their details).
+type GetSaver interface {
```
Comment (stolowski) [inline timings/state.go]: Nice.

File: timings/state.go [hunk 2]
```diff
@@ -116,13 +115,21 @@ func flattenRecursive(data *rootTimingsJSON, timings []*Span, nestLevel int, max
...
+	GetTimings(timings interface{}) error
```
Comment (pedronis) [inline timings/state.go]: I think I should call this one GetMaybeTimings to be consistent with other places that use GetMaybe for when the data being absent is ok (no error)

Comment (stolowski) [reply timings/state.go]: +1

File: overlord/state/timings.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+import (
+	"github.com/snapcore/snapd/timings"
+)
+
+// TimingsForTask creates a new Timings tree for the given task.
+// Returned Timings tree has "task-id", "change-id" and "task-kind"
+// tags set automatically from the respective task.
+func TimingsForTask(task *Task) *timings.Timings {
+	tags := map[string]string{"task-id": task.ID(), "task-kind": task.Kind(), "task-status": task.Status().String()}
```
Comment (bboozzoo) [inline overlord/state/timings.go]:
```suggestion
	tags := map[string]string{
		"task-id":     task.ID(),
		"task-kind":   task.Kind(),
		"task-status": task.Status().String(),
	}
```

File: timings/state.go [hunk 4]
```diff
@@ -142,15 +152,15 @@ func (t *Timings) Save(st *state.State) {
 	if len(stateTimings) > MaxTimings {
 		stateTimings = stateTimings[len(stateTimings)-MaxTimings:]
 	}
-	st.Set("timings", stateTimings)
+	s.SaveTimings(stateTimings)
 }
 
 // Get returns timings for which filter predicate is true and filters out nested timings whose level is greater than maxLevel.
```
Comment (bboozzoo) [inline timings/state.go]: If you happen to push more patches, can you wrap the whole comment?

**Pull request merged**